kernel/nfsft/nfsft.c

```diff
@@ -1250,6 +1257,28 @@ void nfsft_trafo(nfsft_plan *plan)
       /* Use NDFT. */
       nfft_trafo_direct(&plan->plan_nfft);
     }
+    else if (plan->flags & NFSFT_USE_FSFT)
+    {
+      /* Algorithm for equispaced nodes */
+      int N[2];
+      N[0] = 2*plan->N+2;
+      N[1] = 2*plan->N+2;
+
+      for (int j=0; j<N[0]; j++)
+        for (int k=0; k<N[1]; k++)
+          if ((j+k)%2)
+            plan->f_hat_intern[j*N[1]+k] *= -1;
+//	  f_hat[j*N[1]+k] = plan->f_hat_intern[j*N[1]+k] * CEXP(II*KPI*(j+k));
+      fftw_plan plan_fftw = FFTW(plan_dft)(2, N, plan->f_hat_intern, plan->f_hat_intern, FFTW_FORWARD, FFTW_ESTIMATE);
```

Comment:
`fftw_plan_dft` may be used instead of `FFTW(plan_dft)` macro since the current (N)FSFT code only works for double precision.

The FFTW plan and destroy methods are not thread-safe. Please consider using an OMP critical block, see e.g. in `nfft.c` line 6019 and 6218:
```
#ifdef _OPENMP
#pragma omp critical (nfft_omp_critical_fftw_plan)
{
#endif
```